src/config/tenants/tenant.rs

```diff
@@ -246,40 +270,71 @@ async fn display(
         }))
         .await;
 
+        // Track error states for later use
+        let error_states: Vec<bool> = fetched_data
+            .iter()
+            .map(|(_, _, result)| result.is_err())
+            .collect();
+
         let mut display = vec![];
         let mut index = 1;
-
-        for (tenant, realm, result) in fetched_data {
-            let (api_tenant, api_realm) = result?;
-            let is_default = default_tenant.id == tenant.id && default_realm.id == realm.id;
-            let environment = if realm.api_base_url.contains("localhost") {
-                "[l]".to_string()
-            } else {
-                match realm.api_base_url.split('.').last().unwrap_or_default() {
-                    "run" => "[r]".to_string(),
-                    "xyz" => "[s]".to_string(),
-                    "dev" => "[d]".to_string(),
-                    "net" => "[g]".to_string(),
-                    "com" => "".to_string(),
-                    _ => "[?]".to_string(),
+        let mut has_errors = false;
+
+        for (i, (tenant, realm, result)) in fetched_data.into_iter().enumerate() {
+            match result {
+                Ok((api_tenant, api_realm)) => {
+                    let is_default = match &default_tenant_realm {
+                        Some((default_tenant, default_realm)) => {
+                            default_tenant.id == tenant.id && default_realm.id == realm.id
+                        }
+                        None => false,
+                    };
+                    let environment = if realm.api_base_url.contains("localhost") {
+                        "[l]".to_string()
+                    } else {
+                        match realm.api_base_url.split('.').last().unwrap_or_default() {
+                            "run" => "[r]".to_string(),
+                            "xyz" => "[s]".to_string(),
+                            "dev" => "[d]".to_string(),
+                            "net" => "[g]".to_string(),
+                            "com" => "".to_string(),
+                            _ => "[?]".to_string(),
+                        }
+                    };
+                    display.push(RealmDisplay {
+                        index: if is_default {
+                            format!("{}* {}", index, environment)
+                        } else {
+                            format!("{}  {}", index, environment)
+                        },
+                        tenant_name: api_tenant.display_name.clone(),
+                        tenant_id: tenant.id.clone(),
+                        realm_name: api_realm.display_name.clone(),
+                        realm_id: realm.id.clone(),
+                        classification: match api_realm.classification {
+                            Some(Classification::SecureCustomer) => "Secure Customer".to_string(),
+                            Some(Classification::SecureWorkforce) => "Secure Workforce".to_string(),
```

Comment:
We could just implement `ToString` for the Classification enum. but if we are only doing serialization here its not much of an issue.